File: 7-reverse-integer/reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    let xCopy = x;
    x = Math.abs(x)
    let rev = 0
    while(x>0){
        lastDigit = x % 10;
        rev = (rev*10) + lastDigit;
        x = Math.floor(x/10)
    }
    let limit = Math.pow(2,31);
    if(rev < -limit || rev > limit) return 0

    return (xCopy<0) ? -rev : rev;
};

File: 7-reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        revCopy = x
        x = abs(x)
        revNum = 0
        while x > 0:
            lastDigit = x % 10
            revNum = revNum * 10 + lastDigit
            x = x // 10
        
        limit = 2 ** 31
        if revNum < -limit or revNum > limit:
            return 0
        return -revNum if revCopy < 0 else revNum 
